runtime/sema/type.go

```diff
@@ -5021,6 +5022,286 @@ func (t *DictionaryType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Typ
+	elementTypeAnnotationState := r.MemberType.TypeAnnotationState()
+	if elementTypeAnnotationState != TypeAnnotationStateValid {
+		return elementTypeAnnotationState
+	}
+
+	return TypeAnnotationStateValid
+}
+
+func (r *InclusiveRangeType) RewriteWithRestrictedTypes() (Type, bool) {
+	rewrittenElementType, elementTypeRewritten := r.MemberType.RewriteWithRestrictedTypes()
+	if elementTypeRewritten {
+		return &InclusiveRangeType{
+			MemberType: rewrittenElementType,
+		}, true
+	} else {
+		return r, false
+	}
+}
+
+func (t *InclusiveRangeType) BaseType() Type {
+	if t.MemberType == nil {
+		return nil
+	}
+	return &InclusiveRangeType{}
+}
+
+func (t *InclusiveRangeType) Instantiate(typeArguments []Type, report func(err error)) Type {
+	memberType := typeArguments[0]
+	return &InclusiveRangeType{
+		MemberType: memberType,
+	}
+}
+
+func (t *InclusiveRangeType) TypeArguments() []Type {
+	memberType := t.MemberType
+	if memberType == nil {
+		memberType = IntegerType
+	}
+	return []Type{
+		memberType,
+	}
+}
+
+var inclusiveRangeTypeParameter = &TypeParameter{
+	Name:      "T",
+	TypeBound: IntegerType,
+}
+
+func (*InclusiveRangeType) TypeParameters() []*TypeParameter {
+	return []*TypeParameter{
+		inclusiveRangeTypeParameter,
+	}
+}
+
+const InclusiveRangeTypeStartFieldName = "start"
+const inclusiveRangeTypeStartFieldDocString = `
+The start of the InclusiveRange sequence
+`
+const InclusiveRangeTypeEndFieldName = "end"
+const inclusiveRangeTypeEndFieldDocString = `
+The end of the InclusiveRange sequence
+`
+
+const InclusiveRangeTypeStepFieldName = "step"
+const inclusiveRangeTypeStepFieldDocString = `
+The step size of the InclusiveRange sequence
+`
+
+var InclusiveRangeTypeFieldNames = []string{
+	InclusiveRangeTypeStartFieldName,
+	InclusiveRangeTypeEndFieldName,
+	InclusiveRangeTypeStepFieldName,
+}
+
+const InclusiveRangeTypeContainsFunctionName = "contains"
+
+const inclusiveRangeTypeContainsFunctionDocString = `
+Returns true if the given integer is in the InclusiveRange sequence
+`
+
+func (r *InclusiveRangeType) GetMembers() map[string]MemberResolver {
+	r.initializeMemberResolvers()
+	return r.memberResolvers
+}
+
+func InclusiveRangeContainsFunctionType(elementType Type) *FunctionType {
+	return &FunctionType{
+		Parameters: []Parameter{
+			{
+				Label:          ArgumentLabelNotRequired,
+				Identifier:     "element",
+				TypeAnnotation: NewTypeAnnotation(elementType),
+			},
+		},
+		ReturnTypeAnnotation: NewTypeAnnotation(
+			BoolType,
+		),
+	}
+}
+
+func (r *InclusiveRangeType) initializeMemberResolvers() {
+	r.memberResolversOnce.Do(func() {
+		r.memberResolvers = withBuiltinMembers(r, map[string]MemberResolver{
+			InclusiveRangeTypeStartFieldName: {
+				Kind: common.DeclarationKindField,
+				Resolve: func(memoryGauge common.MemoryGauge, identifier string, _ ast.Range, _ func(error)) *Member {
+					return NewPublicConstantFieldMember(
+						memoryGauge,
+						r,
+						identifier,
+						r.ElementType(false),
```

Comment:
Here and for the other fields:

```suggestion
						r.MemberType,
```

Reply:
Done.